packages/palette/src/elements/Text/Text.tsx

```diff
@@ -1,20 +1,27 @@
 import { themeGet } from "@styled-system/theme-get"
+import styled, { css } from "styled-components"
 import { variant } from "styled-system"
-import { styled as primitives } from "../../platform/primitives"
 import { textMixin, TextProps } from "./Text.shared"
 import { TEXT_VARIANTS } from "./tokens"
 
-/**
- * Text
- */
-export const Text = primitives.Text<TextProps>`
+/** Adds ellipsis to overflowing text */
+export const overflowEllipsisMixin = css`
+  overflow: hidden;
+  white-space: nowrap;
+  text-overflow: ellipsis;
+`
+
+/** Text */
+export const Text = styled.div<TextProps & { overflowEllipsis?: boolean }>`
   ${variant({ variants: TEXT_VARIANTS.small })}
+  ${textMixin}
 
   @media (min-width: ${themeGet("breakpoints.0")}) {
     ${variant({ variants: TEXT_VARIANTS.large })}
+    ${textMixin}
```

Comment:
Just PR-creeped this bug fix in. Due to how styled-components emits the media query CSS, we need to double include the textMixin in both contexts so when you specify say a `solid` line-height for Text (which is what I need in the `NavBar`, for instance — a valid use-case, the media query doesn't override it.